client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,64 @@ def build_template(
+                 a version already exists.
+            workfile_creation_enabled (bool): Whether the call is part of
+                creating a new workfile.
+                When True, we only build if the current file is not
+                an existing saved workfile but a "new" file. Basically when
+                enabled we assume the user tries to load it only into a
+                "New File" (unsaved empty workfile).
+                When False, the default value, we assume we explicitly want to
+                build the template in our current scene regardless of current
+                scene state.
 
         """
-        if any(
-            value is None
-            for value in [
-                template_path,
-                keep_placeholders,
-                create_first_version,
-            ]
-        ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
-            if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
-            if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
 
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
-
-        # if first version is created, import template
-        # and populate placeholders
+        # Get default values if not provided
         if (
-            create_first_version
-            and workfile_creation_enabled
-            and created_version_workfile
+            template_path is None
+            or keep_placeholders is None
+            or create_first_version is None
         ):
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
+            if keep_placeholders is None:
+                keep_placeholders: bool = preset["keep_placeholder"]
+            if create_first_version is None:
+                create_first_version: bool = preset["create_first_version"]
+
+        # Build the template if we are explicitly requesting it or if it's
+        # an unsaved "new file".
+        is_new_file = not self.host.get_current_workfile()
+        explicit_build_requested = not workfile_creation_enabled
+        if is_new_file or explicit_build_requested:
+            self.log.info(f"Building the workfile template: {template_path}")
             self.import_template(template_path)
             self.populate_scene_placeholders(
                 level_limit, keep_placeholders)
 
-            # save workfile after template is populated
-            self.save_workfile(created_version_workfile)
+        if not workfile_creation_enabled:
+            # Do not consider saving a first workfile version, if this
```

Comment:
and then I'd probably rephrase this to something more like this:
```
# If we are explicitly requesting to build the template
# we assume it's only doing that and not saving a new workfile
# so we return early
```
* And any reason why you are adding the comment within the scope of the conditional and not before?

Reply:
> And any reason why you are adding the comment within the scope of the conditional and not before?

Merely because it felt related to that condition. I'm fine with it either way, no idea whether there's any style guide for this!